shared/src/commonMain/kotlin/com/example/tasky/login/data/LoginDataSource.kt

```diff
@@ -2,23 +2,35 @@ package com.example.tasky.login.data
 
 import com.example.tasky.common.data.manager.HttpManager
 import com.example.tasky.common.data.model.BaseError
+import com.example.tasky.common.data.util.invalidateBearerTokens
 import com.example.tasky.common.data.util.safeCall
 import com.example.tasky.common.domain.model.ResultWrapper
+import com.example.tasky.common.domain.model.onSuccess
 import com.example.tasky.login.data.model.LoginBody
 import com.example.tasky.login.data.model.LoginResponse
 import com.example.tasky.login.data.model.RegisterBody
 import io.ktor.client.HttpClient
+import io.ktor.client.request.get
 import io.ktor.client.request.post
 import io.ktor.client.request.setBody
 
 class LoginDataSource(
     private val httpClient: HttpClient = HttpManager.httpClient,
 ) {
     suspend fun login(loginBody: LoginBody): ResultWrapper<LoginResponse, BaseError> =
-        safeCall {
+        safeCall<LoginResponse> {
             httpClient.post("/login") {
                 setBody(loginBody)
             }
+        }.onSuccess {
+            httpClient.invalidateBearerTokens()
+        }
+
+    suspend fun logout(): ResultWrapper<Unit, BaseError> =
+        safeCall<Unit> {
+            httpClient.get("/logout")
+        }.onSuccess {
+            httpClient.invalidateBearerTokens()
```

Comment:
There is a bit more to logging out than just invalidating the token. I believe you’ll get to it in a later PR 😄